pkgs/applications/system/tango-controls/default.nix

```diff
@@ -0,0 +1,101 @@
+{ cmake
+, cppzmq
+, doxygen
+, fetchurl
+, graphviz
+, lib
+, libjpeg_turbo
+, libsodium
+, mariadb-connector-c
+, openjdk11
+, pkg-config
+, python3
+, stdenv
+, systemd
+, zeromq
+, zlib
+}:
+let
+  # omniorb is in nixpkgs, but tango doesn't support the latest version
+  omniorb_4_2 = stdenv.mkDerivation rec {
+    pname = "omniorb";
+    version = "4.2.5";
+
+    src = fetchurl {
+      url = "mirror://sourceforge/project/omniorb/omniORB/omniORB-${version}/omniORB-${version}.tar.bz2";
+      sha256 = "1fvkw3pn9i2312n4k3d4s7892m91jynl8g1v2z0j8k1gzfczjp7h";
+    };
+
+    nativeBuildInputs = [ pkg-config ];
+    buildInputs = [ python3 ];
+
+    enableParallelBuilding = true;
+    hardeningDisable = [ "format" ];
+  };
+in
+stdenv.mkDerivation rec {
+  pname = "tango-controls";
+  version = "9.4.2";
+
+  src = fetchurl {
```

Comment:
Why don't you use fetchFromGitLab?

Reply:
Very good question! The thing about tango-controls is that it's assembled from a lot of components, all in different repositories: [starter](https://gitlab.com/tango-controls/starter), [cpptango](https://gitlab.com/tango-controls/cppTango), [tango-database](https://gitlab.com/tango-controls/TangoDatabase).

The `TangoSourceDistribution` repository contains an `ant` build script to clone all these repositories and build them. It also contains a package (the tarball that I download here) that has all these repositories with the proper versions (!) pre-cloned, so building is _much_ easier.